subsys/mgmt/smp_bt.c

```diff
@@ -51,14 +48,12 @@ static ssize_t smp_bt_chr_write(struct bt_conn *conn,
 				const void *buf, u16_t len, u16_t offset,
 				u8_t flags)
 {
-	struct smp_bt_user_data *ud;
 	struct net_buf *nb;
 
 	nb = mcumgr_buf_alloc();
 	net_buf_add_mem(nb, buf, len);
 
-	ud = net_buf_user_data(nb);
-	ud->conn = bt_conn_ref(conn);
+	bt_smp_conn = conn;
```

Comment:
This is broken. You have to use bt_conn_ref() whenever making persistent references to a bt_conn (preferably when making any reference)

Reply:
Also, you're missing the place where you'd release the reference. If you don't get the reference counting right you'll be stuck with a seemingly valid (non-NULL) pointer but you'll never know when it's actually pointing to a valid object. The only way to ensure this is to grab a reference yourself.

Reply:
I wonder whether it is  good idea to take the reference only for offset == 0? AFIK It depends form client side implementation of the long write operation.